dll/win32/shell32/CDefView.cpp

```diff
@@ -577,6 +583,9 @@ void CDefView::UpdateStatusbar()
 
         while ((nItem = m_ListView.GetNextItem(nItem, uFileFlags)) >= 0)
         {
+            if (dwThreadId != m_uUpdateStatusbarThreadId)
+                return;
```

Comment:
Is this done in order to avoid this worker function to be called by two differently launched threads?
If so, in which conditions would there be more than one thread launched?

Reply:
Calculating total size is slow operation and it takes time.
`UpdateStatusbar` is triggered on every selection.
The thread is different from the variable, then the function aborts it.

Reply:
OK. Question: Suppose I am in system32 and I select all the files, then very quickly I cancel the selection (by e.g. clicking somewhere else in the file list) or I change the selection to a different set of selected files (for example, one single file) while staying in that same directory.
Or alternatively, suppose I first select all files in system32, then very quickly open a subbdirectory in system32 (while being in the same explorer window).

--> Would the existing update thread be stopped in a timely fashion, and the status bar reflect the new change?
-

Reply:
A selection change will trigger `LVN_ITEMCHANGED` message and `UpdateStatusbar` call.
Going to another view will trigger `DestroyViewWindow` and it will cancel `UpdateStatusbar` task.

Reply:
I see. Wouldn't it be useful then to stop the previous update thread in UpdateStatusbar() , before starting the new one?

Reply:
The modern computer is very fast. Assigning a handle to `m_hUpdateStatusbarThread` may come after the beginning of the new thread.

Reply:
It would be better to not allow bugs to be present just because "the modern computer" hides them. ReactOS is also run in virtual machines with different speeds, and e.g. running ROS in QEmu on Windows is not the fastest thing in existence. There, you could get into troubles.

Reply:
> Confirmed: Going to another view will trigger DestroyViewWindow and it will cancel UpdateStatusbar task.

To be clear, the hypothetical situation I was presenting in those questions was such that the **SAME** explorer window was being used to go through different subdirectories. Not by killing the window.